### PR TITLE
travis cleanup and split for JS tests

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph 

cc @sravfeyn

### CODE REVIEW

File: .travis/misc-setup.sh [hunk 1]
```diff
@@ -1,3 +1,17 @@
+#!/bin/bash
+set -ev
+
+keytool -genkey \
```
Comment (benrudolph) [inline .travis/misc-setup.sh]: i've seen this when tinkering with travis, do you know what this is for?

File: .travis.yml [hunk 2]
```diff
@@ -3,6 +3,8 @@ sudo: false
 cache:
   directories:
     - $HOME/.cache/pip
+matrix:
+  fast_finish: true
```
Comment (benrudolph) [inline .travis.yml]: cool

**Comment by benrudolph** [comment]
very nice cleanup. should speed up the runs! :athletic_shoe:

File: .travis/misc-setup.sh (hunk 1 above)
Comment (snopoke) [inline .travis/misc-setup.sh]: Its used for jar signing

**Comment by snopoke** [comment]
Not sure why this test is failing
```
======================================================================
ERROR: test_compress_offline (corehq.apps.style.tests.test_compress_command.TestDjangoCompressOffline)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/style/tests/test_compress_command.py", line 8, in test_compress_offline
    call_command('compress', force=True)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 115, in call_command
    return klass.execute(*args, **defaults)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 338, in execute
    output = self.handle(*args, **options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 533, in handle
    return self.handle_noargs(**options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/compressor/management/commands/compress.py", line 274, in handle_noargs
    self.compress(sys.stdout, **options)
  File "/home/travis/virtualenv/local/lib/python2.7/site-packages/compressor/management/commands/compress.py", line 231, in compress
    "%s" % (template.template_name, e))
CommandError: An error occured during rendering /home/travis/build/dimagi/commcare-hq/corehq/apps/dashboard/templates/dashboard/dashboard_new_user.html: 'bootstrap-daterangepicker/daterangepicker.css' could not be found in the COMPRESS_ROOT '/home/travis/build/dimagi/commcare-hq/staticfiles' or with staticfiles.
```

**Comment by benrudolph** [comment]
hey @snopoke i'm guessing that test is failing because the test runner that runs it doesn't run `bower install` before running the tests

File: .travis/matrix-installs.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    nohup bash -c "cd kafka && bin/zookeeper-server-start.sh config/zookeeper.properties &"
+    nohup bash -c "cd kafka && bin/kafka-server-start.sh config/server.properties &"
+    sleep 5
+    kafka/bin/kafka-topics.sh --create --partitions 1 --replication-factor 1 --topic case --zookeeper localhost:2181
+elif [ "${MATRIX_TYPE}" = "javascript" ]; then
+    npm install -g bower
+    npm install -g grunt
+    npm install -g grunt-cli
+    ln -nfs `which bower` /home/travis/bower
+    python manage.py bower install
```
Comment (benrudolph) [inline .travis/matrix-installs.sh]: unfortunately this might have to be in the `python` setup as well for that compression test

Comment (benrudolph) [inline .travis/matrix-installs.sh]: (or alternative you can key off the test runner as well and just install it for that run)

**Comment by snopoke** [comment]
:white-check-mark: thanks fire the tip @benrudolph

**Pull request merged**